### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by John-Arboleda**
# JavaScript capstone project - Your API-based webapp

The JavaScript capstone project is about building a web application based on an external API. The webapp has 2 interfaces: homepage and comment section. In this PR:
#### Homepage
- [x] Build home page interface
- [x] When the page loads, the webapp retrieves data from:
  - [x] The selected API and shows the list of items on the screen.
  - [x] The Involvement API to show the item likes.
- [x] When the user clicks on the Like button of an item, the interaction is recorded in the Involvement API and the screen is updated.
- [x] There is a counter for the number of items (home).
#### Comment
- [x] Build popup with the comment section.
- [x] When you click on the comment button, the web app retrieves data from:
  - [x] The robot API, then the image of the item is displayed in the popup window.
  - [x] The Involvement API to show the comments.
- [x] When the user clicks on the Comment button of a comments section, the name and text is recorded in the Involvement API, and the comment section is updated.
- [x] There is a counter for the number of comments at the top of each comment section.
#### Testing
- [x] Add 3 test cases for the homepage feature.
- [x] Add different test cases for the comment feature.

### CODE REVIEW

File: src/__mocks__/helpers.js [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+
+  </div>
+  <div id="comment-counter"> </div>
+  <ul id="comment-list">
+
+  </ul>
+  <input class="inputName" type="text" placeholder="Your name"></input>
+  <textarea class="inputComment" placeholder="Your insights"></textarea>
+  <button class="add-comment-btn">Comment</button>
+  `;
```
Comment (mikethreels) [inline src/__mocks__/helpers.js:120]: - good job rendering the robots 👍  however Robohash is not called using the async await with the fetch function.

File: src/test/comments.test.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      name: 'Kate',
+      userComment: 'new Comment',
+    }, {
+      id: '2',
+      name: 'Mike',
+      userComment: 'new Comment',
+    },
+    ];
+    expect(commentsTest(commentsArr).length).toBe(2);
+  });
```
Comment (mikethreels) [inline src/test/comments.test.js:23]: - the tests arent't actually testing the functions but only the mock functions, please write the test so they are testing the actual count functions.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._
- [ ] the video has to be shot again as you forgot to address the below points. 
  - All the members in the video should appear in the video and talk about the project.
  - The video contains a highlight about a piece of the code the student finds interesting or is proud of making.
- [ ] when running the server as mentioned in the readme file the application is not working see below screenshot

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AshakaE — APPROVED** [review]
### Hi @John-Arboleda @fabgrel10   🙋‍♀️, 

# STATUS APPROVED ✔
   
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**